packages/discovery-provider/src/utils/multi_provider.py

```diff
@@ -14,12 +14,14 @@ def __init__(self, providers):
         self.providers = [HTTPProvider(provider) for provider in providers.split(",")]
 
     def make_request(self, method, params):
+        last_exception = None
         for provider in random.sample(self.providers, k=len(self.providers)):
             try:
                 return provider.make_request(method, params)
-            except Exception:
+            except Exception as e:
+                last_exception = e
                 continue
-        raise Exception("All requests failed")
+        raise last_exception if last_exception else Exception()
```

Comment:
i think i'm probably reading this wrong... but won't this raise an exception every time?

Reply:
early return on line 20 if any request succeeds, so this just makes it throw a more useful error if nothing succeeds